src/Discord.Net.Core/Utils/ComponentType.cs

```diff
@@ -0,0 +1,8 @@
+namespace Discord.Utils;
+
+public static class ComponentTypeUtils
+{
+    public static bool IsSelectType(this ComponentType type) => type is ComponentType.ChannelSelect
```

Comment:
Is it not far more efficient to add this to the enum itself?

`SelectType = SelectMenu | RoleSelect | UserSelect | MentionableSelect | ChannelSelect`

`if (SelectType.HasFlag(type))`

Open to discussion on this but I do not think this deserves an extension method.

Reply:
Component types aren't flags, so no.